assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -128,107 +290,198 @@ subroutine probit_dist_info(kind, is_state, is_inflation, dist_type, &
-   if(kind == QTY_STATE_VARIABLE) then
-      dist_type = BOUNDED_NORMAL_RH_DISTRIBUTION
-      bounded_below = .false.;    bounded_above = .false.
-      lower_bound   = missing_r8; upper_bound   = missing_r8
-   elseif(kind == QTY_TRACER_CONCENTRATION) then
-      dist_type = BOUNDED_NORMAL_RH_DISTRIBUTION
-      bounded_below = .true.; bounded_above = .false.
-      lower_bound   = 0.0_r8; upper_bound   = missing_r8
-   elseif(kind == QTY_TRACER_SOURCE) then
-      dist_type = BOUNDED_NORMAL_RH_DISTRIBUTION
-      bounded_below = .true.; bounded_above = .false.
-      lower_bound   = 0.0_r8; upper_bound   = missing_r8
-   else
-      write(*, *) 'Illegal kind in obs_error_info'
-      stop
-   endif
+
+      ! Comparing the dist_type in string format to list of potential dist_types
+      dist_type_string = qcf_table_data(QTY_loc(1))%probit_extended_state%dist_type
+      call to_upper(dist_type_string)
+      dist_type_loc = findloc(possible_dist_types, trim(dist_type_string))
+
+      if (dist_type_loc(1) == 0) then
+         write(errstring, *) 'Invalid dist_type: ', trim(dist_type_string)
+         call error_handler(E_ERR, 'probit_dist_info:', errstring, source)
+
+      else
+         dist_type = possible_dist_type_ints(dist_type_loc(1))
+      endif
+
+      bounded_below = qcf_table_data(QTY_loc(1))%probit_extended_state%bounded_below
+      bounded_above = qcf_table_data(QTY_loc(1))%probit_extended_state%bounded_above
+      lower_bound = qcf_table_data(QTY_loc(1))%probit_extended_state%lower_bound
+      upper_bound = qcf_table_data(QTY_loc(1))%probit_extended_state%upper_bound
+
 endif
 
 end subroutine probit_dist_info
 
 !------------------------------------------------------------------------
 
 
-subroutine obs_inc_info(obs_kind, filter_kind, rectangular_quadrature, gaussian_likelihood_tails, &
+subroutine obs_inc_info(obs_qty, filter_kind, rectangular_quadrature, gaussian_likelihood_tails, &
    sort_obs_inc, spread_restoration, bounded_below, bounded_above, lower_bound, upper_bound)
 
-integer,  intent(in)  :: obs_kind
+integer,  intent(in)  :: obs_qty
 integer,  intent(inout) :: filter_kind
 logical,  intent(inout) :: rectangular_quadrature, gaussian_likelihood_tails
 logical,  intent(inout) :: sort_obs_inc
 logical,  intent(inout) :: spread_restoration
 logical,  intent(inout) :: bounded_below, bounded_above
 real(r8), intent(inout) :: lower_bound,  upper_bound
 
+integer :: QTY_loc(1)
+character(len=129) :: qty_name
+
+integer :: filter_kind_loc(1)
+character(len=129), dimension(5) :: possible_filter_kinds
+integer, dimension(5) :: possible_filter_kind_ints
+character(len=129) :: filter_kind_string
+
 ! The information arguments are all intent (inout). This means that if they are not set
 ! here, they retain the default values from the assim_tools_mod namelist. Bounds don't exist 
 ! in that namelist, so default values are set in assim_tools_mod just before the call to here.
```

Comment:
I don't think they need to be anymore. Previous to these changes, the defaults for all these were set in assim_tools_mod. The info for the bounds set right before the call to obs_inc_info (https://github.com/NCAR/DART/blob/6a5fbb6126d472a48a449ab6f13ff671c59bfb41/assimilation_code/modules/assimilation/assim_tools_mod.f90#L986C1-L988C51). I think we can definitely remove these lines linked above from assim_tools_mod and make the bounds info intent(out).

The other control options for obs_inc_info are a bit more difficult. filter_kind, sort_obs_inc,  spread_restoration, rectangular_quadrature, and gaussian_likelihood_tails are being set to the values specified in the &assim_tools_nml section of input.nml, as this comment describes:
([](https://github.com/NCAR/DART/blob/b46a922aacd9f2226d15eccb0de22368196f9078/assimilation_code/modules/assimilation/algorithm_info_mod.f90#L438C1-L443C88)).  

The defaults for these variables are used if they are not specified in the nml ([](https://github.com/NCAR/DART/blob/6a5fbb6126d472a48a449ab6f13ff671c59bfb41/assimilation_code/modules/assimilation/assim_tools_mod.f90#L146C1-L153C48)) 

Then we we get to the call to obs_inc_info, the values are changed to the defaults specified in obs_inc_info OR whatever is set in the qcf table. I think we can still set them to intent(out), but I think it is a bit weird to have our users set these options in both the qcf table and the assim_tools_nml. What do you think @hkershaw-brown ?

Reply:
After thinking more about this about this comment I made, I don't know if its bad that the users can set these options in both the &assim_tools_nml and in the qcf table

While we are setting the values for filter_kind in both the &assim_tools_nml and in the qcf table, the filter_kind read in from the qcf table is specific to observation space increments

But this leads me to a new question, which is if we have a value for filter_kind (or any of the other obs_inc_info options) set in the &assim_tools_nml that is not the default (EAKF = 1), do we want to use this filter_kind in obs_inc_info as the default instead of using EAKF? This would mean that by default, the filter_kind used for the observation space increments would match the filter_kind being used in assim_tools_mod

For example, if a user has filter_kind = 8 () set in their &assim_tools_nml, this would be passed in as the filter_kind argument to obs_inc_info, which would then use this value as the 'default' filter

I could see there being issues where a user sets rectangular_quadrature = .false. or something in the namelist, but they don’t include a qcf table. So it ends up changing the value of rectangular_quatrature to .true. (the default) when it calls obs_inc_info without the user expecting/wanting this to happen